src/main/frontend/components/header.cljs

```diff
@@ -32,25 +37,72 @@
                    (route-handler/go-to-journals!))}
      (ui/icon "home" {:style {:fontSize ui/icon-size}})]))
 
-(rum/defc login
-  [logged?]
-  (when (and (not logged?)
-             (not config/publishing?))
+(rum/defc login < rum/reactive
+  []
+  (let [_ (state/sub :auth/id-token)]
+    (when-not config/publishing?
+      (if (user-handler/logged?)
+        [:span.text-sm.font-medium (user-handler/email)]
 
-    (ui/dropdown-with-links
-     (fn [{:keys [toggle-fn]}]
-       [:a.button.text-sm.font-medium.block {:on-click toggle-fn}
-        [:span (t :login)]])
-     (let [list [{:title (t :login-github)
-                  :url (str config/website "/login/github")}]]
-       (mapv
-        (fn [{:keys [title url]}]
-          {:title title
-           :options
-           {:on-click
-            (fn [_] (set! (.-href js/window.location) url))}})
-        list))
-     nil)))
+        [:a.button.text-sm.font-medium.block {:on-click (fn []
+                                                          (js/window.open "https://logseq-test.auth.us-east-2.amazoncognito.com/oauth2/authorize?client_id=4fi79en9aurclkb92e25hmu9ts&response_type=code&scope=email+openid+phone&redirect_uri=logseq%3A%2F%2Fauth-callback"))}
+         [:span (t :login)]]))))
+
+(rum/defcs file-sync <
+  rum/reactive
+  (rum/local nil ::existed-graphs)
+  [state]
+  (let [_ (state/sub :auth/id-token)
+        sync-state (state/sub :file-sync/sync-state)
+        _ (when sync-state (assert (s/valid? :frontend.fs.sync/sync-state sync-state)))
+        not-syncing? (or (nil? sync-state) (fs-sync/sync-state--stopped? sync-state))
+        *existed-graphs (::existed-graphs state)
+        _ (rum/react file-sync-handler/refresh-file-sync-component)
+        graph-txid-exists? (file-sync-handler/graph-txid-exists?)
+        uploading-files (:current-local->remote-files sync-state)
+        downloading-files (:current-remote->local-files sync-state)]
+    (when-not config/publishing?
+      (when (user-handler/logged?)
+        (when-not (file-sync-handler/graph-txid-exists?)
+          (a/go
+            (let [graphs (a/<! (file-sync-handler/list-graphs))]
```

Comment:
Agree with @tiensonqin that we don't need so much core.async, especially for this case where a simpler promise works fine

Reply:
could u give a recommended implementation here?

Reply:
I'm guessing the [request fn](https://github.com/logseq/logseq/pull/3839#discussion_r829450522) returns a channel. If so, we can't turn one example without changing the core design of the protocols, which is a lot of work and requires more thought and discussion. My comment is more to say that we should be careful with our use of core.async. It's a powerful tool but it has a lot of sharp edges and requires training the team to learn about it. Hopefully we'll use enough of core.async's features that the tradeoff is worth it